proto/osmosis/gamm/v1beta1/genesis.proto

```diff
@@ -3,25 +3,35 @@ package osmosis.gamm;
 
 import "gogoproto/gogo.proto";
 import "google/protobuf/any.proto";
+import "google/protobuf/duration.proto";
 import "cosmos_proto/cosmos.proto";
 import "cosmos/base/v1beta1/coin.proto";
 
 // Params holds parameters for the incentives module
 message Params {
-    repeated cosmos.base.v1beta1.Coin pool_creation_fee = 1 [
-        (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
-        (gogoproto.moretags) = "yaml:\"pool_creation_fee\"",
-        (gogoproto.nullable) = false
-    ];
+  repeated cosmos.base.v1beta1.Coin pool_creation_fee = 1 [
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (gogoproto.moretags) = "yaml:\"pool_creation_fee\"",
+    (gogoproto.nullable) = false
+  ];
+  uint64 num_twap_history_per_deletion = 2;
+  // interval is in the unit of block height
+  uint64 twap_history_deletion_interval = 3;
+  google.protobuf.Duration twap_history_keep_duration = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.stdduration) = true,
+    (gogoproto.jsontag) = "duration,omitempty",
+    (gogoproto.moretags) = "yaml:\"twap_history_keep_duration\""
+  ];
 }
 
-
 option go_package = "github.com/osmosis-labs/osmosis/x/gamm/types";
 
 // GenesisState defines the gamm module's genesis state.
 message GenesisState {
-    repeated google.protobuf.Any pools = 1
-        [ (cosmos_proto.accepts_interface) = "PoolI" ];
-    uint64 next_pool_number = 2;
-    Params params = 3 [ (gogoproto.nullable) = false ];
+  repeated google.protobuf.Any pools = 1
+      [ (cosmos_proto.accepts_interface) = "PoolI" ];
+  uint64 next_pool_number = 2;
+  Params params = 3 [ (gogoproto.nullable) = false ];
+  uint64 next_twap_history_delete_pool_id = 4;
```

Comment:
I'm confused by this as well

Reply:
So basically, every time delete twap history is activated, it only iterates over the designated number of pools. The field `next_twap_history_delete_pool_id` is needed to keep track the id of the pool that deletion should be starting off with